### PR TITLE
Add Motorola Moto G6 (motorola-ali)

### PR DESCRIPTION
**Pull request by julianbraha**
Still not working:
* Camera
* Audio
* Battery data
* Mobile networks

### CODE REVIEW

File: drivers/gpu/drm/panel/Kconfig [hunk 1]
```diff
@@ -141,6 +141,15 @@ config DRM_PANEL_TIANMA_TL052VDXP02
 	  Say Y here if you want to enable support for the Tianma TL052VDXP02
 	  1920x1080 5.2" panel (found on the Motorola Moto G5 variants)
 
+config DRM_PANEL_TIANMA_TL057FVXP01
+	tristate "Tianma TL057FVXP01\" panel"
```
Comment (alikates) [inline drivers/gpu/drm/panel/Kconfig:145]: That escaped quote is intended?

File: drivers/gpu/drm/panel/Kconfig [hunk 2]
```diff
@@ -734,9 +743,9 @@ config DRM_PANEL_HIMAX_HX8399C_FHDPLUS
 	help
 	  Say Y here if you want to enable support for the Himax
 	  HX8399C controller for 1080x1920/2280 LCD panels (found on Xiaomi Mi A2 Lite).
-	  
-	  
-	  
+
+
+
```
Comment (alikates) [inline drivers/gpu/drm/panel/Kconfig:748]: Single line maybe?

File: drivers/gpu/drm/panel/Kconfig (hunk 1 above)
Comment (julianbraha) [reply drivers/gpu/drm/panel/Kconfig]: Thanks, didn't catch that. Removed.

**Comment by alikates** [comment]
Before I merge it, can you fix the conflicts please?

**Comment by julianbraha** [comment]
> Before I merge it, can you fix the conflicts please?

Done.

File: arch/arm64/boot/dts/qcom/sdm450-motorola-ali.dts [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+	};
+};
+
+&pmi8950_wled {
+	status = "okay";
+
+	/* Don't copy-paste! See pmi8950.dtsi */
+
+	qcom,current-limit-microamp = <25000>;
+	qcom,enabled-strings = <0 1 2>;
```
Comment (M0Rf30) [inline arch/arm64/boot/dts/qcom/sdm450-motorola-ali.dts:76]: As far as I know You need only to specify one of these between enabled-strings and num-strings (You should have a warning in the dmesg).

File: arch/arm64/boot/dts/qcom/sdm450-motorola-ali.dts [hunk 4]
```diff
@@ -0,0 +1,291 @@
...
+	qcom,num-strings = <3>;
+
+	qcom,cabc;
+	qcom,eternal-pfet;
```
Comment (M0Rf30) [inline arch/arm64/boot/dts/qcom/sdm450-motorola-ali.dts:80]: it's external, not eternal as reported in 

Documentation/devicetree/bindings/leds/backlight/qcom-wled.yaml

and in

drivers/video/backlight/qcom-wled.c

File: arch/arm64/boot/dts/qcom/sdm450-motorola-ali.dts [hunk 5]
```diff
@@ -0,0 +1,291 @@
...
+};
+
+&dsi0 {
+	panel@0 {
+		compatible = "tianma,tl057fvxp01";
+		reg = <0>;
+
+		backlight = <&pmi8950_wled>;
+
+		supply-names = "vsp", "vsn";
```
Comment (M0Rf30) [inline arch/arm64/boot/dts/qcom/sdm450-motorola-ali.dts:90]: why you specify vsp and vsn regulators if you don't pass them to the panel?
to pass a regulator to the display generator you need to specify the `-r` option.
something like `-r vsp -r vsn`

**Review by M0Rf30 — CHANGES_REQUESTED** [review]

File: arch/arm64/configs/motorola-ali.config [hunk 6]
```diff
@@ -0,0 +1,3 @@
+CONFIG_BACKLIGHT_LED=y
```
Comment (M0Rf30) [inline arch/arm64/configs/motorola-ali.config:1]: Should be CONFIG_BACKLIGHT_QCOM_WLED=y

**Review by M0Rf30 — CHANGES_REQUESTED** [review]

**Comment by julianbraha** [comment]
@M0Rf30 Thanks for reviewing! And apologies for my slow response. I've resolved the issues you brought up.

**Comment by M0Rf30** [comment]
Good as I've seen you don't use vsp and vsn yet. You can remove them if the panel works. they are not used. As I've told you, you need to pass them at panel generation with `-r` flag

**Review by M0Rf30 — CHANGES_REQUESTED** [review]
Good. As I've seen, you don't use vsp and vsn yet.
in the panel you should have something like
```
#include <linux/regulator/consumer.h>
struct tianma_tl057fvxp01 {
	struct drm_panel panel;
	struct mipi_dsi_device *dsi;
	struct gpio_desc *reset_gpio;
	struct regulator_bulk_data supplies[2];
	bool prepared;
};

```
```
static int tianma_tl057fvxp01_probe(struct mipi_dsi_device *dsi)
{
	struct device *dev = &dsi->dev;
	struct tianma_tl057fvxp01 *ctx;
	int ret;

	ctx = devm_kzalloc(dev, sizeof(*ctx), GFP_KERNEL);
	if (!ctx)
		return -ENOMEM;

        ctx->supplies[0].supply = "vsn";
	ctx->supplies[1].supply = "vsp";

	ctx->reset_gpio = devm_gpiod_get(dev, "reset", GPIOD_OUT_HIGH);
	if (IS_ERR(ctx->reset_gpio))
		return dev_err_probe(dev, PTR_ERR(ctx->reset_gpio),
				     "Failed to get reset-gpios\n");

	ctx->dsi = dsi;
	mipi_dsi_set_drvdata(dsi, ctx);

	dsi->lanes = 4;
	dsi->format = MIPI_DSI_FMT_RGB888;
	dsi->mode_flags = MIPI_DSI_MODE_VIDEO | MIPI_DSI_MODE_VIDEO_BURST |
			  MIPI_DSI_CLOCK_NON_CONTINUOUS | MIPI_DSI_MODE_LPM;

	drm_panel_init(&ctx->panel, dev, &tianma_tl057fvxp01_panel_funcs,
		       DRM_MODE_CONNECTOR_DSI);

	ret = drm_panel_of_backlight(&ctx->panel);
	if (ret)
		return dev_err_probe(dev, ret, "Failed to get backlight\n");

	drm_panel_add(&ctx->panel);

	ret = mipi_dsi_attach(dsi);
	if (ret < 0) {
		dev_err(dev, "Failed to attach to DSI host: %d\n", ret);
		drm_panel_remove(&ctx->panel);
		return ret;
	}

	return 0;
}
```

They are automatically generated passing  `-r vsp -r vsn` flags at panel generation. 

And you also done something wrong with the commits (perhaps wrong rebase?). Edits of mine should not appear, only yours

**Comment by julianbraha** [comment]
@M0Rf30 Thanks, that helped. I have re-ported the panel driver (and tested it) with those flags. And also fixed the rebase.

**Review by M0Rf30 — APPROVED** [review]

**Review by alikates — APPROVED** [review]

**Pull request merged**